test/Microsoft.Buld.Sql.Tests/PackTests.cs

```diff
@@ -0,0 +1,172 @@
+                item.AddMetadata("PackagePath", "tools/");
+            });
+
+            // Run dotnet pack
+            string stdOutput, stdError;
+            int exitCode = this.RunDotnetCommand("pack", out stdOutput, out stdError);
+
+            // Verify
+            Assert.AreEqual(0, exitCode, "Pack failed with error " + stdError);
+            Assert.AreEqual(string.Empty, stdError);
+            this.VerifyNugetPackage(test: (PackageArchiveReader package) =>
+            {
+                var files = package.GetFiles();
+                Assert.IsTrue(files.Any(f => f.Equals("content/include_content.txt", StringComparison.OrdinalIgnoreCase)),
+                    "Expected content/include_content.txt to be in the packaged file list.");
+                Assert.IsFalse(files.Any(f => f.Contains("exclude_content.txt", StringComparison.OrdinalIgnoreCase)),
+                    "Expected exclude_content.txt to be excluded from the packaged file list.");
+                Assert.IsTrue(files.Any(f => f.Equals("tools/none.txt", StringComparison.OrdinalIgnoreCase)),
+                    "Expected tools/none.txt to be in the packaged file list.");
+            });
+        }
+
+        /// <summary>
+        /// Verifies Nuget package created by dotnet pack command. Checks for nupkg in bin/configration folder.
+        /// Verifies DACPAC is in the tools folder within the nupkg, and verifies DACPAC is one of its package types.
+        /// </summary>
+        /// <param name="packageNameOverride">When not set, defaults to Name.Version.nupkg</param>
+        /// <param name="test">Additional test to run against the package</param>
+        private void VerifyNugetPackage(string packageNameOverride = "", Action<PackageArchiveReader>? test = null)
+        {
+            // Verify packakge exists
+            string packageName = String.IsNullOrEmpty(packageNameOverride) ? $"{DatabaseProjectName}.1.0.0.nupkg" : packageNameOverride;
+            string packagePath = Path.Combine(this.GetOutputDirectory(), packageName);
+            Assert.IsTrue(File.Exists(packagePath), $"Nuget package not found: {packagePath}");
+
+            using var packageReader = new PackageArchiveReader(packagePath);
+
+            // Verify dacpac is in tools folder
+            var files = packageReader.GetFiles();
+            Assert.IsTrue(files.Any(f => f.Equals($"tools/{DatabaseProjectName}.dacpac", StringComparison.OrdinalIgnoreCase)),
+                $"Expected 'tools/{DatabaseProjectName}.dacpac' to be in the Nuget package.");
+
+            // Verify the DACPAC package type is set
+            var packageTypes = packageReader.GetPackageTypes();
+            Assert.IsTrue(packageTypes.Any(t => t.Name.Equals("DACPAC", StringComparison.OrdinalIgnoreCase)),
+                "Expected 'DACPAC' to be one of the package types.");
+
+            // Run additional test on the package
+            test?.Invoke(packageReader);
```

Comment:
why do we need `?` here?

Reply:
Optional test parameter, default is null